### PR TITLE
Categories CRUD

### PR DESCRIPTION
**Pull request by roblumination**
**Edit categories page**
Now you able to:
- add new _category_ with _subcategories_ and select role
- delete category

**Main page**
Added new mocked post and ability to filter them by category or subcategory

**Firebase services** (supported by @dennissimmo )
Modified Firebace CRUD service:
-  now `getSubCollection` return result with IDs of subcollection;
- added new method to remove subcollection

### CODE REVIEW

File: src/app/features/categories-edit/components/loader/loader.component.scss [hunk 1]
```diff
@@ -0,0 +1,57 @@
+:host {
+	width: 100%;
+	background-color: inherit;
+}
+
+* {
+	transition: all 0.5s, background 0.3s ease-out;
+	// transition-delay: 0.5s, 0s;
```
Comment (Lingsbay) [inline src/app/features/categories-edit/components/loader/loader.component.scss:8]: Delete commented styles if it's unnecessary.

File: src/app/features/categories-edit/pages/categories-list/categories-list.component.scss [hunk 2]
```diff
@@ -1,3 +1,5 @@
+@import '/src/styles/variable.scss';
```
Comment (Lingsbay) [inline src/app/features/categories-edit/pages/categories-list/categories-list.component.scss:1]: @import 'variable.scss'; If you want..

File: src/environments/environment.ts [hunk 3]
```diff
@@ -4,6 +4,25 @@
 
 export const environment = {
 	production: false,
+
+	/**
+	 *
+	 * ORIGINAL CREDENTIALS
+	 * UNCOMENT IT IF I FORGOT :D
```
Comment (Lingsbay) [inline src/environments/environment.ts:11]: Uncoment environment)

File: src/app/features/categories-edit/services/roles.service.ts [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+		return new Promise<void>((resolve, reject) => {
+			const junctionsWriting: Promise<string>[] = [];
+
+			rolesIdToAdd.forEach(roleID => {
+				const currentJunction: RoleCategoryPairDB = {
+					categoryId: categoryID,
+					roleId: roleID
+				};
+				junctionsWriting.push(
+					this.junctionService.addDocWithAutoId(currentJunction)
```
Comment (dennissimmo) [inline src/app/features/categories-edit/services/roles.service.ts:42]: You can use method `addRoleCategoryEntry` method from `roleCategoryFirebase` service

File: src/app/features/categories-edit/services/roles.service.ts [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+	 *	Delets junctions by their IDs
+
+	 * @param junctionsId - array of ID of Category-Role pairs
+	 */
+	deleteJunctionByIds(junctionsId: string[]): Promise<void> {
+		return new Promise<void>((resolve, reject) => {
+			const deletingJunctionsArray: Promise<void>[] = [];
+
+			junctionsId.forEach(juncID => {
+				deletingJunctionsArray.push(this.junctionService.deleteDoc(juncID));
```
Comment (dennissimmo) [inline src/app/features/categories-edit/services/roles.service.ts:66]: Please, add method deleteRoleCategoryEntry(recordId: string) to `roleCategoryFirebase` service, he should simply call deleteDoc() method. Then you can call newly created method here

File: src/app/features/main/pipes/post-search.pipe.ts [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+	name: 'postSearch'
+})
+export class PostSearchPipe implements PipeTransform {
+	transform(posts: Post[], categoryID: string, subCategoryId: string): Post[] {
+		if (categoryID == 'all') {
+			return posts;
+		} else if (subCategoryId) {
+			// console.log('clicked on subcat');
+			return posts.filter(post => post.subCategory == subCategoryId);
+		} else {
```
Comment (dennissimmo) [inline src/app/features/main/pipes/post-search.pipe.ts:14]: You can omit `else` statement

**Review by dennissimmo — APPROVED** [review]
Good job!

**Review by dennissimmo — APPROVED** [review]
Approved

**Review by Lingsbay — APPROVED** [review]
Good work.

**Review by Sithipio — APPROVED** [review]
Fine

**Pull request merged**